server/manager/src/main/java/org/apache/accumulo/manager/upgrade/UpgradeCoordinator.java

```diff
@@ -179,15 +186,22 @@ public synchronized Future<Void> upgradeMetadata(ServerContext context,
           .submit(() -> {
             try {
               for (int v = currentVersion; v < AccumuloDataVersion.get(); v++) {
-                log.info("Upgrading Root from data version {}", v);
-                upgraders.get(v).upgradeRoot(context);
+                log.info("Upgrading Root - current version {} as step towards target version {}", v,
+                    AccumuloDataVersion.get());
+                if (upgraders.get(v) != null) {
+                  upgraders.get(v).upgradeRoot(context);
+                }
```

Comment:
I think you should replace lines 191-1932 with `upgraders.get(v).upgradeRoot(context);`. You added the `requireNonNull` check in the commit you mentioned below, so it shouldn't be null. If it null here, then this code will hide it and move on.

Reply:
5436f22281  changes the null check to match  the ZooKeeper  sectionn - using rquireNonNull - while the check may be redundant if the previous check(s) pass it isolates issues with execution order.  The reason behind the check is to provide a better error message - during development a misconfiguration was throwing a null pointer and having a log statement helped.  

Originally, the code skipped that section if null and it was pointed out by @ctubbsii that it would be better to fail.